specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -351,6 +852,261 @@
+      "type": "object",
+      "properties": {
+        "accessRulesVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Version number that increases with every update to access rules within the profile."
+        }
+      },
+      "description": "Properties of NSP profile."
+    },
+    "NspAccessRule": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NspAccessRuleProperties",
+          "description": "Properties of the NSP access rule."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the access rule that is unique within a profile. This name can be used to access the resource."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "NSP access rule identifier."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "The NSP access rule resource"
+    },
+    "NspAccessRuleListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NspAccessRule"
+          },
+          "description": "Gets a page of NSP access rule"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next page of results."
+        }
+      },
+      "description": "Result of the request to list NSP access rules. Contains a list of NSP access rules and a URL link to get the next set of results."
+    },
+    "NspAccessRuleProperties": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the scope assignment resource."
+        },
+        "direction": {
+          "type": "string",
+          "enum": [
+            "Inbound",
+            "Outbound"
+          ],
+          "x-ms-enum": {
+            "name": "AccessRuleDirection",
+            "modelAsString": true
+          },
+          "description": "Direction that specifies whether the access rules is inbound/outbound."
+        },
+        "addressPrefixes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Inbound address prefixes (IPv4/IPv6)"
+        },
+        "fullyQualifiedDomainNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Outbound rules fully qualified domain name format."
+        },
+        "subscriptions": {
```

Comment:
>subscriptions [](http://example.com/codeflow?start=9&length=13)

rename to subscriptionResourceIds

Reply:
We had used this in the ARM linked access checks. Changing the property name would need a manifest change. We will consider this change before moving to the next preview with more customers. It is currently targeted for a single customer.

Reply:
This tends to be an issue for people writing arm templates m if the name is not descriptive to explain if it expects the name or the id. I would highly recommend to change , but it is up to your team to decide.